src/analyses/memLeak.ml

```diff
@@ -20,6 +20,105 @@ struct
   let context _ d = d
 
   (* HELPER FUNCTIONS *)
+  let get_global_vars () =
+    (* Filtering by GVar seems to account for declarations, as well as definitions of global vars *)
+    List.filter_map (function GVar (v, _, _) -> Some v | _ -> None) !Cilfacade.current_file.globals
+
+  let get_global_struct_ptr_vars () =
+    List.filter_map (function GVar (v, _, _) -> Some v | _ -> None) !Cilfacade.current_file.globals
+    |> List.filter (fun v -> match v.vtype with TPtr (TComp _, _) | TPtr ((TNamed ({ttype = TComp _; _}, _)), _) -> true | _ -> false)
```

Comment:
This isn't very reliable, there may be more than one `typedef`. To do it properly, matching should be on `Cil.unrollType`.

Reply:
Should be fixed in https://github.com/goblint/analyzer/pull/1241/commits/6cc01b5177b6bc31899c290b6ae65660bf4aa805